docker/driver.go

```diff
@@ -0,0 +1,99 @@
+package docker
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/bblfsh/performance"
+
+	"bitbucket.org/creachadair/shell"
+	"github.com/creachadair/staticfile"
+	"gopkg.in/src-d/go-errors.v1"
+	"gopkg.in/src-d/go-log.v1"
+)
+
+//go:generate go run github.com/creachadair/staticfile/compiledata -pkg docker -out static.go conf/drivers.json
+
+// InstallDriver downloads driver's repository, checkouts to a given commit,
+// builds docker image and installs driver's image to bblfshd container.
+// Requires bblfshd container running.
+func InstallDriver(language, commit string) (func(), error) {
+	data, err := staticfile.ReadAll("conf/drivers.json")
+	if err != nil {
+		return nil, errors.NewKind("failed to read drivers config").Wrap(err)
+	}
+
+	drivers := make(map[string]string)
+	if err := json.Unmarshal(data, &drivers); err != nil {
+		return nil, err
+	}
+	log.Debugf("Supported drivers: %v", drivers)
+
+	driver, ok := drivers[language]
+	if !ok {
+		return nil, errors.NewKind("language %v is not supported").New(language)
+	}
+	log.Debugf("Selected driver: %v", driver)
+
+	image := driverImage(driver, commit)
+	closer, err := downloadAndBuildDriver(driver, commit, image)
+	if err != nil {
+		return closer, err
+	}
+	if err := installDriverToBblfshd(bblfshdContainer, language, image); err != nil {
+		return closer, err
+	}
+
+	return closer, err
+}
+
+func downloadAndBuildDriver(driver, commit, image string) (func(), error) {
+	tmpDir, err := ioutil.TempDir("", driver)
+	if err != nil {
+		return nil, err
+	}
+	log.Debugf("Created temp directory %v", tmpDir)
+
+	closer := func() {
```

Comment:
[minor, optional nit]: I suggest the name `cleanup` instead of `closer` since the latter sounds like `io.Closer`, but the function signature isn't the same (no error).

Reply:
Done